crates/libcontainer/src/utils.rs

```diff
@@ -312,12 +315,94 @@ where
     unreachable!("retry loop completed without returning a result.");
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum NetDevicesError {
+    #[error("unable to move network devices without a NET namespace")]
+    NoNetNamespace,
+    #[error("network devices are not supported in rootless containers")]
+    RootlessNotSupported,
+    #[error("invalid network device name: {0}")]
+    InvalidDeviceName(String),
+    #[error(transparent)]
+    IO(#[from] std::io::Error),
+    #[error(transparent)]
+    Spec(#[from] MissingSpecError),
+}
+
+// check if given spec is valid for netDevices
+pub fn validate_spec_for_net_devices(
+    spec: &Spec,
+    syscall: &dyn Syscall,
+) -> Result<(), NetDevicesError> {
+    let linux = spec
+        .linux()
+        .as_ref()
+        .ok_or(NetDevicesError::Spec(MissingSpecError::Linux))?;
+
+    if linux.net_devices().is_none() {
+        return Ok(());
+    }
+
+    if linux
+        .namespaces()
+        .as_ref()
+        .unwrap()
```

Comment:
Is this always safety?

Reply:
fix: https://github.com/youki-dev/youki/pull/3163/commits/0fe0e20796e53f42271df24c9366bc209edba2af